Dockerfile

```diff
@@ -1,9 +1,15 @@
 # Builder
-
 FROM golang:1.12.6-alpine3.9 as builder
 
+RUN apk --no-cache add ca-certificates git
+
 WORKDIR /go/src/github.com/openbank/openbank
 
+ENV GO111MODULE=on
+COPY go.mod .
+COPY go.sum .
+RUN go mod download
```

Comment:
why not run `go mod download` just after `copy . .`? you won't need to copy go.mod and go.sum

Reply:
ahh, thank you
updated

Reply:
i decide to copy go.mod and go.sum before copying all project files. This because, when we copy go.mod and go.sum together with all the project file,  **go mod download** command will always download all dependencies while we make changes to the code. this will only re-use the downloaded dependencies for exactly the same build. 
Copying the go.mod and go.sum before all files, will check the dependencies from docker cache, and will download dependencies **ONLY IF** go.mod and go.sum change.

Reply:
What novian writes makes sense. But please include a clear comment; otherwise it's going to be confusing to everyone else.

Also, this can be simpler:

* Don't use GOPATH at all. Use something dummy like `/src` as the workdir
* You don't need to set GO111MODULE if you're outside GOPATH
* You can merge the two COPY lines into one, like `COPY go.* .`

Reply:
updated @mvdan